### PR TITLE
overwrite flag added to az storage blob upload-batch

### LINKED ISSUE #10724: New azure-cli requires --overwrite to replace canary
Noticed in chartmuseum test failure:
```
Pushing binaries to Azure
ERROR: The specified blob already exists.
...
ErrorCode:BlobAlreadyExists
If you want to overwrite the existing one, please add --overwrite in your command.
```

Chartmuseum copies script from Helm to upload canary artifacts.

In this script we need to add an `--overwrite` flag to `az storage blob upload-batch`:
https://github.com/helm/helm/blob/main/.circleci/deploy.sh#L49

CircleCI builds on `main` will fail.

### PR DESCRIPTION
**Pull request by sourikghosh**
Signed-off-by: Sourik Ghosh <sourikghosh31@gmail.com>

**What this PR does / why we need it**:
adds a overwrite flag to az storage blob upload-batch to replace canary.
closes #10724

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by yxxhero** [comment]
/lgtm
Thanks your PR.

**Comment by sourikghosh** [comment]
@jdolitsky can you have a look at this

**Review by jdolitsky — CHANGES_REQUESTED** [review]
I was thinking about this. The new behavior actually protects from supply chain attacks.

Could you modify this to only add `--overwrite` if `[[ "${VERSION}" == "canary" ]]` ?

File: .circleci/deploy.sh [hunk 1]
```diff
@@ -46,4 +46,8 @@ make build-cross
 make dist checksum VERSION="${VERSION}"
 
 echo "Pushing binaries to Azure"
+if [[ "${VERSION}" == "canary" ]]; then
+az storage blob upload-batch -s _dist/ -d "$AZURE_STORAGE_CONTAINER_NAME" --pattern 'helm-*' --connection-string "$AZURE_STORAGE_CONNECTION_STRING" --overwrite
+else
 az storage blob upload-batch -s _dist/ -d "$AZURE_STORAGE_CONTAINER_NAME" --pattern 'helm-*' --connection-string "$AZURE_STORAGE_CONNECTION_STRING"
+if
```
Comment (jdolitsky) [inline .circleci/deploy.sh:53]: last thing, can we format this?
```
if [[ "${VERSION}" == "canary" ]]; then
  az storage blob upload-batch -s _dist/ -d "$AZURE_STORAGE_CONTAINER_NAME" --pattern 'helm-*' --connection-string "$AZURE_STORAGE_CONNECTION_STRING" --overwrite
else
  az storage blob upload-batch -s _dist/ -d "$AZURE_STORAGE_CONTAINER_NAME" --pattern 'helm-*' --connection-string "$AZURE_STORAGE_CONNECTION_STRING"
if

Comment (sourikghosh) [reply .circleci/deploy.sh:53]: ohhh snap,,, yeah yeah

**Review by jdolitsky — APPROVED** [review]

**Comment by jdolitsky** [comment]
thanks @sourikghosh !

**Pull request merged**